gitpod-io/gitpod
# [server] Set JWT cookie on sign-in

## Description


* New JWTs have a `keyId` (`kid`) property in the header indicating the ID of the key used to create the token
* When validating, we read the `kid` property from the JWT header, and use it to lookup the corresponding Public Key for verification

#### Preview status

gitpod:summary

## Related Issue(s)

* Depends on https://github.com/gitpod-io/gitpod/pull/17200
* Depends on https://github.com/gitpod-io/gitpod/pull/17219

## How to test

1. Preview
2. Sign in
3. Look at cookies, take the value of the `*_jwt_` cookie and put it into https://jwt.io/ and decode it. See the header contains the `keyid` property with `0001`

## Release Notes

```release-note
NONE
```

## Documentation


## Build Options:

      Run the build with werft instead of GHA
      Run Leeway with `--dont-test`

<details>
<summary>Publish Options</summary>

</details>

<details>
<summary>Installer Options</summary>

  Add desired feature flags to the end of the line above, space separated
</details>

#### Preview Environment Options:
      If enabled this will build `install/preview`
      If enabled this will create the environment on GCE infra
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

/hold
